### PR TITLE
Update climo

### PR DESCRIPTION
**Pull request by doutriaux1**
@gleckler1 @durack1

### CODE REVIEW

**Comment by doutriaux1** [comment]
I need to flake8 it. But it's usable.

**Comment by gleckler1** [comment]
@doutriaux1  CD got her working without too much difficulty yea!   Currently it only saves one file ("ann") , even calculating seasons for "all".   Using CMOR we of course have to have a separate file for each season.    The next big step to make is **being able to loop over models and variables**.   I first tried that using glob in the param file and sending list to "files", but then I realized "files" is reserved for cases where multiple files need to be linked via cdscan to produce a climatology.  Its useful to have that option but most of the time we'll be dealing with xmls that already span files in time.   

The most common usage (with model data) will be to be able to provide from the parameter file a list of vars (for VAR below) and a list of mods for (MODEL) below.   Anyhow, I think its close!

/work/cmip5/historical/atm/mo/VAR/cmip5.MODEL.historical.r1i1p1.mo.atm.Amon.VAR.ver-*.xml

**Comment by doutriaux1** [comment]
@gleckler1 will ook at this on Thursday, @zshaheen I will come to you to see if we can use the -d here. I think it's a perfect fit.

**Comment by gleckler1** [comment]
@doutriaux1 @zshaheen   It would be great to try out the -d option for this routine application.   All I need is an example to run with.   Any hints?

**Comment by doutriaux1** [comment]
@gleckler1 see: https://e3sm-project.github.io/acme_diags/docs/html/quick-guide-edison.html#advanced-running-custom-diagnostics

**Comment by doutriaux1** [comment]
Actully check there: https://e3sm-project.github.io/acme_diags/docs/html/defining-parameters.html

**Comment by gleckler1** [comment]
@doutriaux1  I'm new to .cfg ... can the contents be python?  What I need to construct is something like:  
modsnames = ['GFDL-CM3', 'NCAR-CESM1']
variables = ['tas','pr']
modpath = /work/cmip5/historical/atm/mo/%(variable)/cmip5.%(model).historical.r1i1p1.mo.atm.Amon.%(variable).ver-*.xml

**Comment by doutriaux1** [comment]
@gleckler1 no they're json files as far as I understand it.

**Comment by doutriaux1** [comment]
@gleckler1 my bad they are python files looks like @zshaheen can you comment?

**Comment by doutriaux1** [comment]
@gleckler1 I think the idea is that with the `.cfg` we do not need to create the loop in the driver. It is automatically `created` for you by creating a bunch of processes running independently on multiple processes. It makes the driver easier to implement/maintain and read. @zshaheen do I understand this correctly?

**Comment by doutriaux1** [comment]
@gleckler1 the difference was that in pcmdi_metrics we need all the results to be stored in the same output file, so there was potential concurrency issues. Here everything is cleanly separated.

**Comment by gleckler1** [comment]
@doutriaux1 @zshaheen  OK, I'm game to give it a go, but could use help setting up a simple example... it could take me days to figure that out; once I have a sample in hand I can run with it...

**Comment by doutriaux1** [comment]
yeah.. Will try but can't really take a look until Thu... @zshaheen any chance you can jump start @gleckler1

**Comment by zshaheen** [comment]
@gleckler1 @doutriaux1 Charles idea is correct. Let's talk this week about creating an example if you all feel the need.

@gleckler1 A `cfg` file can only have **non-programmatic things**. So no adding two variables together, and dictionaries do not work. However, you can have floats, ints, strings, and lists of all of these things. We use the `ConfigParser` module, so inputs are limited by what that supports.

This is why for `e3sm_diags`, we have inputs via `py` and `cfg` files. The programatic stuff can go into the `py` file. For us, all of the changes between runs are non-programmatic things.

**Comment by gleckler1** [comment]
@doutriaux1 No can workey... need .cfg example better yet complete -p and -d ...

**Comment by doutriaux1** [comment]
@gleckler1 I need to complete the parallel driver tomorrow and it should have examples. I moved away from -d it's not what we want in the end.

**Comment by gleckler1** [comment]
@doutriaux1  Is CD taking Thur off?

File: examples/cmip_clims_param_nocmor.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+
+vars = ['pr','rlut', 'rsut','rsutcs','rlutcs','tas','prw','tauu','tauv','uas','vas','psl','hus','ta','ua','va', 'zg']
+vars = ['rlut']
+
+realm = 'atm'
+MIP = 'cmip5'
+exp = 'historical'
+
+modpath = '/work/cmip5-test/' + exp + '/' + realm + '/mo_new/'
```
Comment (durack1) [inline examples/cmip_clims_param_nocmor.py]: @doutriaux1 do we want to change this to `os.path.join()`?

File: examples/parameter_files/cmip_clims_param.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import glob
+
+vars = ['pr','rlut', 'rsut','rsutcs','rlutcs','tas','prw','tauu','tauv','uas','vas','psl','hus','ta','ua','va', 'zg']
+#vars = ['rlut']
+var = ['rlut']
+realm = 'atm'
+MIP = 'cmip5'
+exp = 'historical'
+
+modpath = '/work/cmip5-test/' + exp + '/' + realm + '/mo_new/'
```
Comment (durack1) [inline examples/parameter_files/cmip_clims_param.py]: @doutriaux1 do we want to change this to os.path.join()?

File: examples/cmip_clims_param_nocmor.py (hunk 1 above)
Comment (doutriaux1) [reply examples/cmip_clims_param_nocmor.py]: @durack1 actually the _nocmor stuff needs to go away

File: examples/parameter_files/cmip_clims_param.py (hunk 2 above)
Comment (doutriaux1) [reply examples/parameter_files/cmip_clims_param.py]: @durack1 yes will do this thx!

**Comment by gleckler1** [comment]
@doutriaux1 I'm ready to give this a pre-merge test run... did you say you created a notebook to demonstrate how this works?

**Comment by gleckler1** [comment]
@doutriaux1 never mind, I see its already merged... will try out now

**Comment by gleckler1** [comment]
@doutriaux1  Can't get example param to work and don't understand enough to debug it:

 pcmdi_compute_climatologies.py -p cmip_clims_param.py
.
.
.
historical/atm/mo/rlut/cmip5.CESM1-WACCM.historical.r1i1p1.mo.atm.Amon.rlut.ver-v20130314.latestX.xml', '/work/cmip5/historical/atm/mo/rlut/cmip5.CESM1-WACCM.historical.r1i1p1.mo.atm.Amon.rlut.ver-v20120606.latestX.xml', '/work/cmip5/historical/atm/mo/rlut/cmip5.MPI-ESM-MR.historical.r1i1p1.mo.atm.Amon.rlut.ver-1.latestX.xml', '/work/cmip5/historical/atm/mo/rlut/cmip5.CESM1-BGC.historical.r1i1p1.mo.atm.Amon.rlut.ver-v20130213.latestX.xml']
modpath is  /work/cmip5/historical/atm/mo
Traceback (most recent call last):
  File "/export/gleckler1/anaconda2/envs/pmp_cdat8/bin/pcmdi_compute_climatologies.py", line 125, in <module>
    if not os.path.exists(A.file):
  File "/export/gleckler1/anaconda2/envs/pmp_cdat8/lib/python2.7/genericpath.py", line 26, in exists
    os.stat(path)
TypeError: coercing to Unicode: need string or buffer, list found
(pmp_cdat8) bash-4.1$

**Comment by doutriaux1** [comment]
sounds like you have py3

**Comment by doutriaux1** [comment]
let me fix it for py3 as well

**Comment by gleckler1** [comment]
@doutriaux1 nope I'm on 2.7.   I did have to modify sample para to access data (see attached)
[cmip_clims_param.py.txt](https://github.com/PCMDI/pcmdi_metrics/files/2199241/cmip_clims_param.py.txt) with the .txt so that GH would take it.

**Comment by gleckler1** [comment]
@doutriaux1 Still not able to use this climo.   It runs, and seems to be spawning the appropriate number of jobs, but only the last "file" is saved.  
[cmip_clims_param.py.txt](https://github.com/PCMDI/pcmdi_metrics/files/2210792/cmip_clims_param.py.txt)

**Comment by doutriaux1** [comment]
@gleckler1 ok let me take a look at your param file

**Comment by gleckler1** [comment]
@doutriaux1 thnks... I've just confirmed there are multiple models in "file", but the processes spawned in the tmp param files all point to the same model.

**Pull request merged**